lib/jsdom/browser/Window.js

```diff
@@ -34,8 +34,17 @@ const jsGlobals = require("./js-globals.json");
 const GlobalEventHandlersImpl = require("../living/nodes/GlobalEventHandlers-impl").implementation;
 const WindowEventHandlersImpl = require("../living/nodes/WindowEventHandlers-impl").implementation;
 
-exports.createWindow = function (options) {
-  return new Window(options);
+let beforeParse;
+const beforeParseDefaultOptions = { isMain: false };
+
+exports.setBeforeParse = function (cb) {
+  beforeParse = cb;
+};
+
+exports.createWindow = function (options, beforeParseOptions) {
+  const window = new Window(options);
+  beforeParse(window, Object.assign({}, beforeParseDefaultOptions, beforeParseOptions));
```

Comment:
There's no need for this `Object.assign()`; you can just pass `beforeParseOptions` directly.